api/base/serializers.py

```diff
@@ -13,6 +12,27 @@
 from api.base.exceptions import InvalidQueryStringError, Conflict
 
 
+def format_relationship_links(related_link, self_link, rel_meta, self_meta):
+
+    ret = {
+        'links': {
+            'related': {
+                'href': related_link,
+                'meta': rel_meta
+            },
+            'self': {
+                'href': self_link,
+                'meta': self_meta
+            }
+        }
+    }
+
+    # Not currently returning self links
+    if not ret['links']['self']['href']:
```

Comment:
After reading [here](http://jsonapi.org/format/#document-resource-object-relationships), [here](http://jsonapi.org/format/#document-resource-object-linkage) and glancing at the [JSON Schema](http://jsonapi.org/schema) again, it appears that the following is valid:

```
"relationships": {
    "parent": {
        "data": null
    }
}
```

because `/relationships/parent/data` is a [resource linkage](http://jsonapi.org/format/#document-resource-object-linkage), and resource linkages may be `null`.

However, this is invalid:

```
"relationships": {
    "parent": null
}
```

I am fine with excluding the relationship object for both null and non-existent relationships for now, at least until we encounter a need to disambiguate null vs. non-existent.